### ISSUE TITLE
After upload, change of view causes segfault

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#2647]
**[Submitted to the original trac issue database at 11.18am, Thursday, 21st January 2010]**

Immediately after uploading, I try to zoom out using mouse wheel:

```
#0  0x00007ffff59d7390 in QPainterPath::QPainterPath(QPainterPath const&) ()
   from /usr/lib/libQtGui.so.4
#1  0x000000000057858e in Road::getPath (this=0x287a190) at Maps/Road.cpp:597
#2  0x00000000005b6e46 in MapView::drawBackground (this=0xee79c0, theP=...) at MapView.cpp:428
#3  0x00000000005b7baa in MapView::updateStaticBackground (this=0xee79c0) at MapView.cpp:748
#4  0x00000000005b9bba in MapView::paintEvent (this=0xee79c0, anEvent=0x7fffffffc8c0)
    at MapView.cpp:194
#5  0x00007ffff5900196 in QWidget::event(QEvent*) () from /usr/lib/libQtGui.so.4
#6  0x00000000005b382e in MapView::event (this=0xee79c0, event=0x7fffffffc8c0) at MapView.cpp:1129
#7  0x00007ffff58b001d in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
   from /usr/lib/libQtGui.so.4
#8  0x00007ffff58b807a in QApplication::notify(QObject*, QEvent*) () from /usr/lib/libQtGui.so.4
#9  0x00007ffff50c0c9c in QCoreApplication::notifyInternal(QObject*, QEvent*) ()
   from /usr/lib/libQtCore.so.4
#10 0x00007ffff59071be in QWidgetPrivate::drawWidget(QPaintDevice*, QRegion const&, QPoint const&, int, QPainter*, QWidgetBackingStore*) () from /usr/lib/libQtGui.so.4
#11 0x00007ffff5a7ce6e in ?? () from /usr/lib/libQtGui.so.4
#12 0x00007ffff58f9220 in QWidgetPrivate::syncBackingStore() () from /usr/lib/libQtGui.so.4
#13 0x00007ffff5900031 in QWidget::event(QEvent*) () from /usr/lib/libQtGui.so.4
#14 0x00007ffff5c82d2b in QMainWindow::event(QEvent*) () from /usr/lib/libQtGui.so.4
#15 0x00007ffff58b001d in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
   from /usr/lib/libQtGui.so.4
#16 0x00007ffff58b807a in QApplication::notify(QObject*, QEvent*) () from /usr/lib/libQtGui.so.4
#17 0x00007ffff50c0c9c in QCoreApplication::notifyInternal(QObject*, QEvent*) ()
   from /usr/lib/libQtCore.so.4
#18 0x00007ffff50c18e4 in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) ()
   from /usr/lib/libQtCore.so.4
#19 0x00007ffff50e97d3 in ?? () from /usr/lib/libQtCore.so.4
#20 0x00007ffff298213a in g_main_context_dispatch () from /lib/libglib-2.0.so.0
#21 0x00007ffff2985998 in ?? () from /lib/libglib-2.0.so.0
#22 0x00007ffff2985b4c in g_main_context_iteration () from /lib/libglib-2.0.so.0
#23 0x00007ffff50e939c in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>)
    () from /usr/lib/libQtCore.so.4
#24 0x00007ffff5946f1f in ?? () from /usr/lib/libQtGui.so.4
#25 0x00007ffff50bf562 in QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
   from /usr/lib/libQtCore.so.4
#26 0x00007ffff50bf934 in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) ()
   from /usr/lib/libQtCore.so.4
#27 0x00007ffff50c1ba4 in QCoreApplication::exec() () from /usr/lib/libQtCore.so.4
#28 0x0000000000620ee5 in main (argc=1, argv=0x7fffffffe548) at Main.cpp:208

```

**Comment by tomhughes** [comment]
**Author: Koying**
**[Added to the original trac issue at 1.27pm, Tuesday, 26th January 2010]**

Did you receive messages during the upload?

**Comment by tomhughes** [comment]
**Author: tms13**
**[Added to the original trac issue at 11.03am, Wednesday, 17th February 2010]**

Only the normal messages.
I'm no longer seeing this issue, so I'm withdrawing it.  It was probably fixed in passing.

**Comment by tomhughes** [comment]
**Author: Koying**
**[Added to the original trac issue at 11.23pm, Wednesday, 17th February 2010]**

I found a bug which might be related to this one. It involves coastlines and the crash happens after they are removed, thru a delte or a "file-new"

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 11.25pm, Wednesday, 17th February 2010]**

(In [20049]) FIX : Crash after coastlines are removed from the document (fixes #2647)
